src/realm/string_interner.cpp

```diff
@@ -426,8 +428,11 @@ std::optional<StringID> StringInterner::lookup(StringData sd)
         return {};
     }
     std::lock_guard lock(m_mutex);
-    if (sd.data() == nullptr)
-        return 0;
+    if (sd.data() == nullptr) {
+        if (null_seen)
```

Comment:
I guess this is the machinery for handling a null string, right?

Reply:
yup.